src/sql/src/plan/statement/dml.rs

```diff
@@ -955,10 +959,11 @@ GROUP BY mlm.global_id, mlm.lir_id, mse.worker_id"#,
                 "megsa.levels AS levels",
                 "megsa.to_cut AS to_cut",
                 "megsa.hint AS hint",
-                "pg_size_pretty(savings) AS savings",
+                "pg_size_pretty(megsa.savings) AS savings",
             ]);
             from.extend(["JOIN mz_introspection.mz_dataflow_global_ids mdgi ON (mlm.global_id = mdgi.global_id)",
-            "LEFT JOIN mz_introspection.mz_expected_group_size_advice megsa ON (megsa.dataflow_id = mdgi.id AND mlm.operator_id_start <= megsa.region_id AND megsa.region_id < mlm.operator_id_end)"]);
+            "LEFT JOIN (generate_series((mlm.operator_id_start) :: int8, (mlm.operator_id_end - 1) :: int8) AS valid_id JOIN \
```

Comment:
I'm wondering if this generate_series could be on the left side of the left join join. Having it on the right side means that the right side is correlated with the left side, which is a complicated left join lowering case, with its own code path.

Reply:
I started with it on the left side, but I got redundant rows. (There may be a good way to hide those, but my SQL skill wasn't sufficient to the task.) The resulting plan is not _small_, but there is no cross join...